src/core/server/elasticsearch/client/configure_client.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { stringify } from 'querystring';
+import { Client } from '@elastic/elasticsearch';
+import { Logger } from '../../logging';
+import { parseClientOptions, ElasticsearchClientConfig } from './client_config';
+
+export const configureClient = (
+  config: ElasticsearchClientConfig,
+  { logger, scoped = false }: { logger: Logger; scoped?: boolean }
+): Client => {
+  const clientOptions = parseClientOptions(config, scoped);
+
+  const client = new Client(clientOptions);
+  addLogging(client, logger, config.logQueries);
+
+  return client;
+};
+
+const addLogging = (client: Client, logger: Logger, logQueries: boolean) => {
+  client.on('response', (err, event) => {
+    if (err) {
+      logger.error(`${err.name}: ${err.message}`);
```

Comment:
are we okay not to log `warnings` anymore? 
https://github.com/elastic/kibana/blob/0a09aee0a519f8daf42f4d27f0aa776a9e4b3051/src/core/server/elasticsearch/legacy/elasticsearch_client_config.ts#L195

Reply:
I took https://github.com/elastic/kibana/pull/69905#discussion_r445524499 as a 'yes', as warning are per-request and can now be handled by the consumers. but we can decide to log them on our own. I don't really have an opinion on that one.

Reply:
I’m wondering if the client should log warnings more aggressively, maybe via `process.emitWarning`.
This is because I think users will never proactively go and read the `warnings` key. What do you think?

Reply:
`process.emitWarning` seems way too aggressive imho. Being able to get them using the event emitter API is fine.